main/src/ca/uwaterloo/flix/language/phase/Inliner1.scala

```diff
@@ -0,0 +1,986 @@
+    * Add corresponding symbol from `symbols` to substitution map `env0`, mapping old symbols to fresh symbols.
+    */
+  private def bind(exp0: OccurrenceAst1.Expr, symbols: List[(OccurrenceAst1.FormalParam, Occur)], args: List[OutExpr], varSubst0: VarSubst, subst0: Subst, inScopeSet0: InScopeSet, context0: Context)(implicit root: OccurrenceAst1.Root, flix: Flix): MonoAst.Expr = {
+    def bnd(syms: List[(OccurrenceAst1.FormalParam, Occur)], as: List[OutExpr], env: VarSubst): MonoAst.Expr = (syms, as) match {
+      case ((_, occur) :: nextSymbols, e1 :: nextExpressions) if isDeadAndPure(occur, e1.eff) =>
+        // If the parameter is unused and the argument is pure, then throw it away.
+        bnd(nextSymbols, nextExpressions, env)
+
+      case ((_, occur) :: nextSymbols, e1 :: nextExpressions) if isDead(occur) =>
+        // If the parameter is unused and the argument is NOT pure, then put it in a statement.
+        val nextLet = bnd(nextSymbols, nextExpressions, env)
+        val eff = Type.mkUnion(e1.eff, nextLet.eff, e1.loc)
+        MonoAst.Expr.Stm(e1, nextLet, nextLet.tpe, eff, exp0.loc)
+
+      case ((OccurrenceAst1.FormalParam(sym, _, _, _, _), _) :: nextSymbols, e1 :: nextExpressions) =>
+        val freshVar = Symbol.freshVarSym(sym)
+        val env1 = env + (sym -> freshVar)
+        val nextLet = bnd(nextSymbols, nextExpressions, env1)
+        val eff = Type.mkUnion(e1.eff, nextLet.eff, e1.loc)
+        MonoAst.Expr.Let(freshVar, e1, nextLet, nextLet.tpe, eff, exp0.loc)
+
+      case _ => visitExp(exp0, varSubst0 ++ env, subst0, inScopeSet0, context0)
+    }
+
+    bnd(symbols, args, Map.empty)
+  }
+
+  private def freshFormalParam(fp0: OccurrenceAst1.FormalParam)(implicit flix: Flix): (MonoAst.FormalParam, VarSubst) = fp0 match {
+    case OccurrenceAst1.FormalParam(sym, mod, tpe, src, loc) =>
+      val freshVarSym = Symbol.freshVarSym(sym)
+      val subst = Map(sym -> freshVarSym)
+      (MonoAst.FormalParam(freshVarSym, mod, tpe, src, loc), subst)
+  }
+
+  private def refreshFormalParam(fp0: MonoAst.FormalParam)(implicit flix: Flix): (MonoAst.FormalParam, VarSubst) = fp0 match {
+    case MonoAst.FormalParam(sym, mod, tpe, src, loc) =>
+      val freshVarSym = Symbol.freshVarSym(sym)
+      val subst = Map(sym -> freshVarSym)
+      (MonoAst.FormalParam(freshVarSym, mod, tpe, src, loc), subst)
+  }
+
+  private def refreshFormalParams(fparams0: List[MonoAst.FormalParam])(implicit flix: Flix): (List[MonoAst.FormalParam], VarSubst) = {
+    val (fps, substs) = fparams0.map(refreshFormalParam).unzip
+    val subst = substs.reduceLeft(_ ++ _)
+    (fps, subst)
+  }
+
+  private def refreshBinders(expr0: OutExpr)(implicit subst0: VarSubst, flix: Flix): OutExpr = expr0 match {
+    case MonoAst.Expr.Cst(cst, tpe, loc) =>
+      Expr.Cst(cst, tpe, loc)
+
+    case MonoAst.Expr.Var(sym, tpe, loc) =>
+      val freshVarSym = subst0.getOrElse(sym, sym)
```

Comment:
From the other code i'd have expected `subst0.getOrElse(sym, Symbol.freshVarSym ...)`. Is this your temporary method to check that names are not an issue?